src/main/scala/org/allenai/pipeline/Workflow.scala

```diff
@@ -46,14 +48,21 @@ object Node {
         !persisted.artifact.exists
       case _ => false
     }
-    Node(stepInfo.className,
+    val timeTaken = step match {
+      case producer: Producer[_] => producer.timeTaken map (_.toMillis)
```

Comment:
Yes, I prefer this check here, so that the the dependencies can stay having type `PipelineStep`